DesignSafe-CI/DS-User-Guide
# feat: DES-3045 allow non-flat nav in mkdocs.yml

## Overview

Allow pages to be listed in navigation manually at deeper than 2 levels. Do not let old nav bugs resurface.

## Related

- [DES-3045](https://tacc-main.atlassian.net/browse/DES-3045)
- requires https://github.com/TACC/TACC-Docs/pull/42

## Changes

- **add** minimal overwrite of `nav.html` to fix most bugs
- **add** minimal script to fix another bug

## Testing

1. Checkout branch.
2. `git cherry-pick 02d8621d`
    <sub>This (reverted) commit unflattens Use Cases navigation.</sub>
3. Extensively test navigation of "Use Cases".
4. Verify navigation "works".
5. Verify none of these bugs occur:
    - [DES-2493](https://tacc-main.atlassian.net/browse/DES-2493)
    - [DES-2468](https://tacc-main.atlassian.net/browse/DES-2468)
    - [TUP-489](https://tacc-main.atlassian.net/browse/TUP-489)

## UI

> [!IMPORTANT]
> This only works if tested [via Docker](https://github.com/DesignSafe-CI/DS-User-Guide#b-via-docker). Testing [via Python](https://github.com/DesignSafe-CI/DS-User-Guide#b-via-docker) will have bugs.

<details><summary>Before</summary>

https://github.com/user-attachments/assets/fe6ec021-2e13-4334-9d2c-32df0036bdab

https://github.com/user-attachments/assets/00e93c7c-18ef-4881-b427-e39b57efbae8

</details>

<details open><summary>After</summary>

https://github.com/user-attachments/assets/c101710b-dac6-4d4f-b3b0-213a9f9fd814

https://github.com/user-attachments/assets/bda5ba4a-7501-4d2d-a21b-56208c1df6e0

</details>
